components/menu/Menu.js

```diff
@@ -17,162 +17,191 @@ const POSITION = {
 };
 
 class Menu extends PureComponent {
-  state = {
-    active: this.props.active,
-  };
+  state = {};
 
   componentDidMount() {
-    this.positionTimeoutHandle = setTimeout(() => {
-      const { width, height } = this.menuNode.getBoundingClientRect();
-      const position = this.props.position === POSITION.AUTO ? this.calculatePosition() : this.props.position;
-      this.setState({ position, width, height });
-    });
+    const { width, height } = this.menuNode.getBoundingClientRect();
+    this.setState({ width, height });
+
+    const { position } = this.props;
+    if (position === POSITION.AUTO) {
+      this.setState({ position: this.calculatePosition });
+    }
   }
 
-  componentWillReceiveProps(nextProps) {
-    if (this.props.position !== nextProps.position) {
-      const position = nextProps.position === POSITION.AUTO ? this.calculatePosition() : nextProps.position;
-      this.setState({ position });
-    }
-
-    /**
-     * If the menu is going to be activated via props and its not active, verify
-     * the position is appropriated and then show it recalculating position if its
-     * wrong. It should be shown in two consecutive setState.
-     */
-    if (!this.props.active && nextProps.active && !this.state.active) {
-      if (nextProps.position === POSITION.AUTO) {
-        const position = this.calculatePosition();
-        if (this.state.position !== position) {
-          this.setState({ position, active: false }, () => {
-            this.activateTimeoutHandle = setTimeout(() => {
-              this.show();
-            }, 20);
-          });
-        } else {
-          this.show();
-        }
-      } else {
-        this.show();
-      }
+  static getDerivedStateFromProps(props, state) {
+    const newStateSlice = {};
+
+    if (props.active !== state.active) {
+      newStateSlice.active = props.active;
+    }
+
+    if (props.position !== state.position) {
+      newStateSlice.position = props.position;
     }
 
-    /**
-     * If the menu is being deactivated via props and the current state is
-     * active, it should be hid.
-     */
-    if (this.props.active && !nextProps.active && this.state.active) {
+    if (newStateSlice.active || newStateSlice.position) {
+      return newStateSlice;
+    }
+
+    return null;
```

Comment:
Q: Why do you want to specifically return null?

Reply:
To indicate that no state should be updated, it is specified in the (React docs)[https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops].

Reply:
Aha. I was wondering why they wouldn't allow `undefined` because that's what a function returns, but I guess it's because you might say `return object` while `object` is undefined... But then again, that would also be fine because then it would be the same result as we have now by returning `null` 🤔 ... Not sure why they chose this... But idd it's in the docs 🤷‍♂️